src/Microsoft.NetCore.Analyzers/Core/Security/UseSharedAccessProtocolHttpsOnly.cs

```diff
@@ -0,0 +1,140 @@
+            {
+                var microsoftWindowsAzureStorageNamespaceSymbol = compilationStartAnalysisContext
+                            .Compilation
+                            .GlobalNamespace
+                            .GetMembers("Microsoft")
+                            ?.FirstOrDefault()
+                            .GetMembers("WindowsAzure")
+                            .OfType<INamespaceOrTypeSymbol>()
+                            .FirstOrDefault()
+                            .GetMembers("Storage")
+                            .OfType<INamespaceOrTypeSymbol>()
+                            .FirstOrDefault();
+
+                if (microsoftWindowsAzureStorageNamespaceSymbol == null)
+                {
+                    return;
+                }
+
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+
+                compilationStartAnalysisContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
+                {
+                    var owningSymbol = operationBlockStartContext.OwningSymbol;
+                    var interproceduralAnalysisConfig = InterproceduralAnalysisConfiguration.Create(
+                                                            operationBlockStartContext.Options,
+                                                            SupportedDiagnostics,
+                                                            defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.None,
+                                                            cancellationToken: operationBlockStartContext.CancellationToken,
+                                                            defaultMaxInterproceduralMethodCallChain: 1);
+
+                    operationBlockStartContext.RegisterOperationAction(operationAnalysisContext =>
+                    {
+                        var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                        var methodSymbol = invocationOperation.TargetMethod;
+
+                        if (methodSymbol.Name != "GetSharedAccessSignature")
+                        {
+                            return;
+                        }
+
+                        var namespaceSymbol = methodSymbol.ContainingNamespace;
+
+                        if (namespaceSymbol == null)
+                        {
+                            return;
+                        }
+
+                        var namespaceQualifiedName = namespaceSymbol.ToDisplayString();
+
+                        if (namespaceQualifiedName != "Microsoft.WindowsAzure.Storage" &&
+                            !namespaceQualifiedName.StartsWith("Microsoft.WindowsAzure.Storage.", StringComparison.Ordinal))
+                        {
+                            return;
+                        }
+
+                        var valueContentAnalysisResultOpt = ValueContentAnalysis.GetOrComputeResult(
```

Comment:
>valueContentAnalysisResultOpt [](start = 28, length = 29)

Nit: No need for "Opt" in the variable name.

Also, move the GetOrComputeResult() call to the innermost if, to avoid unnecessary DFA.

Reply:
+1 avoid DFA until as late as possible